src/components/package-port/CodeAndPreview.tsx

```diff
@@ -390,9 +392,80 @@ export function CodeAndPreview({ pkg }: Props) {
+                  )
+                // Check for manual edit conflicts
+                if (
+                  pcb_manual_edit_conflict_warnings.length > 0 ||
+                  schematic_manual_edit_conflict_warnings.length > 0
+                ) {
+                  const warnings = [
+                    ...(pcb_manual_edit_conflict_warnings || []),
+                    ...(schematic_manual_edit_conflict_warnings || []),
+                  ]
+
+                  warnings.forEach((warning) => {
+                    const isSchematicWarning =
+                      warning.type === "schematic_manual_edit_conflict_warning"
+                    toast({
+                      title: !isSchematicWarning
+                        ? "Pcb Manual Edit Conflicts"
+                        : "Schematic Manual Edit Conflicts",
+                      description: (
+                        <div className="flex items-center justify-between gap-2">
+                          <span>{warning.message}</span>
+                          <div className="flex flex-col w-full mt-2 mb-1">
+                            <button
+                              className="flex items-center gap-2 px-3 py-1.5 bg-white text-amber-700 hover:bg-white/90 rounded-md cursor-pointer font-medium transition-colors duration-200 w-full justify-center border border-white"
+                              onClick={() =>
+                                handleCustomEditsRemoval(
+                                  state.pkgFilesWithContent.reduce(
+                                    (acc, file) => ({
+                                      ...acc,
+                                      [file.path]: file.content,
+                                    }),
+                                    {},
+                                  ),
+                                  (filename, content) => {
+                                    setState((prev) => ({
+                                      ...prev,
+                                      pkgFilesWithContent:
+                                        prev.pkgFilesWithContent.map((f) =>
+                                          f.path === filename
+                                            ? { ...f, content }
+                                            : f,
+                                        ),
+                                    }))
+                                  },
+                                  toast,
+                                  findTargetFile(
+                                    state.pkgFilesWithContent,
+                                    null,
+                                  )?.path,
+                                  isSchematicWarning,
+                                )
+                              }
+                            >
+                              <AlertTriangleIcon className="h-4 w-4" />
+                              <span>Autofix</span>
+                            </button>
```

Comment:
can we isolate some of this to a separate function? `CodeAndPreview` historically has grown very large